falcon/app.py

```diff
@@ -886,3 +887,16 @@ def _get_body(self, resp, wsgi_file_wrapper=None):
             return iterable, None
 
         return [], 0
+
+
+# NOTE(mikeyusko): This function is temporary and will be deleted once
+# the API alias will be removed
+def api(*args, **kwargs):
+    warnings.warn(
```

Comment:
There is a potential issue with vanilla `warnings` they are actually suppressed under older Python versions. IIRC CPython 3.7 rectifies this (?), but we still support 3.5/3.6.

Cf the approach we are taking for other deprecations in Falcon: https://github.com/falconry/falcon/blob/master/falcon/util/misc.py#L59

Reply:
@vytas7 Nice catch! will fix it